ironfish-rust-nodejs/src/frost.rs

```diff
@@ -7,45 +7,75 @@ use crate::{
     to_napi_err,
 };
 use ironfish::{
-    frost::{keys::KeyPackage, round2::Randomizer, SigningPackage},
+    frost::{
+        keys::KeyPackage, round1::SigningCommitments, round2, round2::Randomizer, SigningPackage,
+    },
     frost_utils::{
-        signature_share::create_signature_share as create_signature_share_rust,
-        signing_commitment::{
-            create_signing_commitment as create_signing_commitment_rust, SigningCommitment,
-        },
+        signature_share::SignatureShare, signing_commitment::SigningCommitment,
         split_spender_key::split_spender_key,
     },
     participant::{Identity, Secret},
     serializing::{bytes_to_hex, fr::FrSerializable, hex_to_bytes, hex_to_vec_bytes},
     SaplingKey,
 };
+use ironfish_frost::nonces::deterministic_signing_nonces;
 use napi::{bindgen_prelude::*, JsBuffer};
 use napi_derive::napi;
 use rand::thread_rng;
+use std::ops::Deref;
+
+fn try_deserialize_signers<I, S>(signers: I) -> Result<Vec<Identity>>
+where
+    I: IntoIterator<Item = S>,
+    S: Deref<Target = str>,
```

Comment:
should we have some unit tests for this function + edgecases?

Reply:
So... I tried to add some tests, but unfortunately it looks like `ironfish-rust-nodejs` is untestable. Running `cargo test` on the crate results in linking errors (which is understandable, given the purpose of the crate, but this makes testing impossible with the current state of things). Maybe in the long term we can move this function somewhere else where it can be tested.

FWIW, the underlying serialization code (`Identity::deserialize_from`) does have tests.